# snpArcher -- standalone-qc branch

Howdy y'all,  the `standalone-qc` branch of Eric's fork of this project just has a few little modifications
that Eric C. Anderson made so that the super-awesome, quick-and-dirty qc module in the
snpArcher workflow can be run on any VCF.gz or BCF file that you happen to have, without
having to run the whole snpArcher pipeline. It is also setup so that it should be easy
to process multiple VCF files (different species, different reference genomes) with only
minimal configuration overhead (preparing a csv file with the paths to the VCF and the `.fai`
files.

This branch has a directory, `standalone-qc-example` that has a vcf.gz
file in it with 16 chinook salmon at 4 pieces of chromosome, and then some scaffolds, which
should be useful for testing and demonstration.

To test this out on your system, the steps are:

1.  Clone this fork of the snpArcher repo:
    ```sh
    # this is using the SSH address
    git clone git@github.com:eriqande/snpArcher.git
    ```
2.  `cd` into the resulting snpArcher repo and switch to the `standalone-qc` branch:
    ```sh
    cd snpArcher
    git checkout standalone-qc
    ```
3.  Dry-run the standalone qc example by running the Snakefile in the `workflow/qc` directory, like this:
    ```sh
    snakemake -np --cores 8 --use-conda -s workflow/modules/qc/Snakefile --configfile standalone-qc-example/config.yaml
    ```
    It is probably worth explaining that line a little.  It is doing a dry-run `-n` and printing
    out the shell commands `-p`, allowing for up to 8 cores of use, and it is using conda for
    managing software.  The Snakefile is specified using the `-s` option to be
    `workflow/modules/qc/Snakefile` and the config file for this run is in
    `standalone-qc-example/config.yaml`.
    





# snpArcher

<img src="./docs/img/logo.png" alt="snpArcher logo" height="300"/>


snpArcher is a reproducible workflow optimized for nonmodel organisms and comparisons across datasets, built on the [Snakemake](https://snakemake.readthedocs.io/en/stable/index.html#) workflow management system. It provides a streamlined approach to dataset acquisition, variant calling, quality control, and downstream analysis.

### Usage
For usage instructions and complete documentation, please visit our [docs](https://snparcher.readthedocs.io/en/latest/).

### Datasets generated by snpArcher
A number of resequencing datasets have been run with snpArcher generating consistent variant calls, available via [Globus](https://www.globus.org/) in the [Comparative Population Genomics Data collection](https://app.globus.org/file-manager?origin_id=a6580c44-09fd-11ee-be16-195c41bc0be4&origin_path=%2F). Details of data processing are described [in our manuscript](https://www.biorxiv.org/content/10.1101/2023.06.22.546168v1). If you use any of these datasets in your projects, please cite both the [snpArcher paper](https://www.biorxiv.org/content/10.1101/2023.06.22.546168v1) and the original data producers.

### Citing snpArcher
- Cade D Mirchandani, Allison J Shultz, Gregg W C Thomas, Sara J Smith, Mara Baylis, Brian Arnold, Russ Corbett-Detig, Erik Enbody, Timothy B Sackton, A fast, reproducible, high-throughput variant calling workflow for population genomics, Molecular Biology and Evolution, 2023;, msad270, https://doi.org/10.1093/molbev/msad270
- Also, make sure to cite the tools you used within snpArcher.
